parquet/parquet.cpp

```diff
@@ -1356,7 +1356,7 @@ static void add_null(pq_column *c, int16_t r, int16_t d)
     ss << "column '" << c->n->name << "' is required";
     throw pq::ParquetException(ss.str());
   }
-  update_levels(c, r, d);
+  update_levels(c, r, d - 1 ? d > 0 : 0);
```

Comment:
My understanding (reflected by tools like `parquet_reader` and `parquet-dump`) is that the way to encode nulls is to use the definition level for the next level up. So using the test schema here ( https://github.com/washort/lua_sandbox_extensions/blob/3a1d6b63b5605905f7b09b02ae923de8b5ac26cb/parquet/tests/test.lua#L246-L270 ) an input like `{my_map = {foo = 1, bar = parquet.null, blee = 2}}` results in these parquet columns:

```
BINARY my_map.key_value.key 
----------------------------------------------------------------------------------------------------------------------------------------
*** row group 1 of 1, values 1 to 3 *** 
value 1: R:0 D:1 V:blee
value 2: R:1 D:1 V:bar
value 3: R:1 D:1 V:foo

INT32 my_map.key_value.value 
----------------------------------------------------------------------------------------------------------------------------------------
*** row group 1 of 1, values 1 to 3 *** 
value 1: R:0 D:2 V:2
value 2: R:1 D:1 V:<null>
value 3: R:1 D:2 V:1
```

That's from `parquet-dump`, here's the `parquet_reader_` output: https://gist.github.com/washort/41a31c71adf5899d1e2d5976218d4a30